http3/http_stream.go

```diff
@@ -75,12 +84,33 @@ func (s *stream) Read(b []byte) (int, error) {
 				return 0, err
 			}
 			switch f := frame.(type) {
-			case *headersFrame:
-				// skip HEADERS frames
-				continue
 			case *dataFrame:
 				s.bytesRemainingInFrame = f.Length
 				break parseLoop
+			case *headersFrame:
+				if s.conn.perspective == protocol.PerspectiveServer {
+					continue
+				}
+				if f.Length > s.maxHeaderBytes {
+					return 0, fmt.Errorf("HEADERS frame too large: %d bytes (max: %d)", f.Length, s.maxHeaderBytes)
+				}
+				p := make([]byte, f.Length)
+				s.Stream.Read(p)
+				trailers, err := s.decoder.DecodeFull(p)
+				if err != nil {
+					// Ignoring invalid frame
+					continue
+				}
+
+				for _, trailer := range trailers {
+					if s.resp.Trailer == nil {
+						s.resp.Trailer = http.Header{}
+					}
+					s.resp.Trailer.Add(trailer.Name, trailer.Value)
+				}
+
+				// Trailer Frame is the last frame.
+				return 0, nil
```

Comment:
There is questions in the previous PR about if this is the best decision. I'm also not 100% sure on this. We may instead want to set a flag to ensure we don't accept data frames after the trailer frame.